mypy/plugins/singledispatch.py

```diff
@@ -9,18 +9,13 @@
     FunctionLike
 )
 from mypy.plugin import CheckerPluginInterface, FunctionContext, MethodContext, MethodSigContext
-from typing import Dict, List, NamedTuple, Optional, Sequence, TypeVar, Union
-from typing_extensions import Final, TypedDict
-
-SingledispatchInfo = TypedDict('SingledispatchInfo', {
-    'fallback': CallableType,
-    # dict of dispatch type to the registered function
-    # dispatch type is stored separately from function because the dispatch type might be different
-    # from the type of the first argument if a type is passed as an argument to register
-    # TODO: use Instance for register type because register requires register type to be subclass
-    # of type
-    'registered': Dict[Type, CallableType],
-})
+from typing import List, NamedTuple, Optional, Sequence, TypeVar, Union
+from typing_extensions import Final
+
+SingledispatchTypeVars = NamedTuple('SingledispatchTypeVars', [
```

Comment:
You can use class-based NamedTuples now.

Reply:
Judging from https://travis-ci.com/github/python/mypy/jobs/520283005, mypyc doesn't seem to support class-based NamedTuples yet (which seems to be related to mypyc/mypyc#861). I'm going to revert this for now, and hopefully we can go back to the class-based syntax when that gets fixed.